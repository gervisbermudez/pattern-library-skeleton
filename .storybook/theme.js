import {create} from '@storybook/theming/create'

const theme = create({
  base: 'light',
  brandTitle: '#zapatillasFromMarsğŸ‘ŸğŸš€',
  brandUrl: 'https://github.com/klaufel/pattern-library-skeleton'
})

export default theme
