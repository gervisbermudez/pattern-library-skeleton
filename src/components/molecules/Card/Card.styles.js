import styled from 'styled-components'

export const Wrapper = styled.div`
  max-width: 640px;
  margin: 80px auto;
  padding: ${props => props.theme.spacing.md};
  border-radius: ${props => props.theme.radius.sm};
  box-shadow: ${props => props.theme.shadows.sm};
`
