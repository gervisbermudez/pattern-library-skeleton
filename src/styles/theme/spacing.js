import tokens from '../../../tokens/json/spacing'

const spacing = {
  xxs: tokens.spacing.xxs.value,
  xs: tokens.spacing.xs.value,
  sm: tokens.spacing.sm.value,
  md: tokens.spacing.md.value,
  lg: tokens.spacing.lg.value,
  xl: tokens.spacing.xl.value,
  xxl: tokens.spacing.xxl.value
}

export default spacing
