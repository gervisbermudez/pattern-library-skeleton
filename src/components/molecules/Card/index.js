import React from 'react'
import PropTypes from 'prop-types'
import {Wrapper} from './Card.styles'
import Heading from '../../atoms/Heading'

const Card = ({title, children}) => (
  <Wrapper>
    {title && (
      <Heading as="h3" size="big">
        {title}
      </Heading>
    )}
    {children}
  </Wrapper>
)

Card.displayName = 'Card'

Card.propTypes = {
  /** Title to displayin the card */
  title: PropTypes.string,
  /** Content to be included in the card */
  children: PropTypes.node.isRequired
}

export default Card
