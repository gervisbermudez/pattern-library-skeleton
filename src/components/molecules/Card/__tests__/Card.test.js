import React from 'react'
import 'jest-styled-components'
import {render} from '@testing-library/react'
import {ThemeProvider} from 'styled-components'
import theme from '../../../../styles/theme'
import Card from '..'

describe('Card', () => {
  test('it works default', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <Card>
          <p>Lorem ipsum dolor</p>
        </Card>
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with title', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <Card title="Card title">
          <p>Lorem ipsum dolor</p>
        </Card>
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
