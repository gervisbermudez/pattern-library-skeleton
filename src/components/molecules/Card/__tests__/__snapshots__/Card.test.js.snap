// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card it works default 1`] = `
.c0 {
  max-width: 640px;
  margin: 80px auto;
  padding: 24px;
  border-radius: 8px;
  box-shadow: 0px 4px 8px rgba(0,0,0,0.25);
}

<div
  class="c0"
>
  <p>
    Lorem ipsum dolor
  </p>
</div>
`;

exports[`Card it works with title 1`] = `
.c0 {
  max-width: 640px;
  margin: 80px auto;
  padding: 24px;
  border-radius: 8px;
  box-shadow: 0px 4px 8px rgba(0,0,0,0.25);
}

.c1 {
  font-size: 24px;
  line-height: 28px;
  font-weight: 700;
  margin: 0 0 24px;
  font-size: 28px;
  line-height: 32px;
  font-weight: 700;
}

<div
  class="c0"
>
  <h3
    class="c1"
  >
    Card title
  </h3>
  <p>
    Lorem ipsum dolor
  </p>
</div>
`;
